langserver/langserver.py

```diff
@@ -151,8 +151,28 @@ def usages(script, parent_span):
                                           "Script.usages"):
             return script.usages()
 
+    def setenv_from_initialization_options(self, params):
+        '''Sets environment variables from initializationOptions. This sets
+        environment variables for the entire process. In the future, this
+        might be modified to manage environment variables per workspace.'''
+
+        if "initializationOptions" not in params:
+            return
+        initOptions = params["initializationOptions"]
+        if not isinstance(initOptions, dict):
+            return
+        if "env" not in initOptions:
+            return
+        envVars = initOptions["env"]
+        if not isinstance(envVars, dict):
+            return
```

Comment:
It's not great that this is just silent if the user does the wrong thing. I would think you should throw an exception if they do. Then you can write much clearer code:

```python
opts = params.get('initializationOptions', {})
for k, v in opts.get('env', {}):
  ...
```